sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java

```diff
@@ -474,13 +478,18 @@ Mono<Void> closeAsync(AmqpShutdownSignal shutdownSignal) {
         });
 
         return Mono.whenDelayError(
-            cbsCloseOperation.doFinally(signalType ->
-                logger.atVerbose()
-                    .addKeyValue(SIGNAL_TYPE_KEY, signalType)
-                    .log("Closed CBS node.")),
-            managementNodeCloseOperations.doFinally(signalType ->
-                logger.atVerbose()
-                    .log("Closed management nodes.")))
+                cbsCloseOperation.doFinally(signalType ->
+                    logger.atVerbose()
+                        .addKeyValue(SIGNAL_TYPE_KEY, signalType)
+                        .log("Closed CBS node.")),
+                managementNodeCloseOperations.doFinally(signalType ->
+                    logger.atVerbose()
+                        .addKeyValue(SIGNAL_TYPE_KEY, signalType)
+                        .log("Closed management nodes.")),
+                emitShutdownSignalOperation.doFinally(signalType ->
```

Comment:
Emit shutdown signal after CBS closing function returns, instead of waiting CBS completely closed. Therefore, the ReactorReceiver(s) can be closed in parallel with CBS node.

Reply:
Thank you Kun! As you explained earlier, I understand the new shutdown order will help solve "the aggressive log" problem. 

But I wonder if there is a different way to solve the "the aggressive log" without changing the shutdown signal order? 

[For the new shutdown order, I'm confident in the explicit-close case, but we need to carefully evaluate the side effects in the server initiated closure (implicit-close) and recovery cases. It involves more thinking given the permutations service can do it. We could treat the "close timeout" as a separate issue following that evaluation]

So localizing the discussion to "the aggressive log" - Is it possible to solve it with minimal impact? For example instead of `takeUntilOther(shutdownSignalSink.asMono())`, does ````.repeat(() -> !this.isDisposed());```` works?

\\cc @conniey

Reply:
That makes sense. I would feel comfortable using repeat(() -> this.isDisposed())

Reply:
Thanks Anu and Connie. Yes, the `.repeat(() -> !this.isDisposed())` can solve the "the aggressive log" problem. 

You are right, I haven't considered the server-initiated closure, I'll check the code to see if we could summarize some for future test.

For "close timeout" problem, I'll create a separated issue to follow up.  Maybe have the no-wait time as you mentioned, instead of changing the closing order.

Reply:
Thanks Kun.